commit 8aa525a9340da4227797a06221ca08399006635f
Author: James Chapman <jchapman@katalix.com>
Date:   Mon Mar 21 18:10:25 2011 -0700

    l2tp: fix possible oops on l2tp_eth module unload
    
    A struct used in the l2tp_eth driver for registering network namespace
    ops was incorrectly marked as __net_initdata, leading to oops when
    module unloaded.
    
    BUG: unable to handle kernel paging request at ffffffffa00ec098
    IP: [<ffffffff8123dbd8>] ops_exit_list+0x7/0x4b
    PGD 142d067 PUD 1431063 PMD 195da8067 PTE 0
    Oops: 0000 [#1] SMP
    last sysfs file: /sys/module/l2tp_eth/refcnt
    Call Trace:
     [<ffffffff8123dc94>] ? unregister_pernet_operations+0x32/0x93
     [<ffffffff8123dd20>] ? unregister_pernet_device+0x2b/0x38
     [<ffffffff81068b6e>] ? sys_delete_module+0x1b8/0x222
     [<ffffffff810c7300>] ? do_munmap+0x254/0x318
     [<ffffffff812c64e5>] ? page_fault+0x25/0x30
     [<ffffffff812c6952>] ? system_call_fastpath+0x16/0x1b
    
    Signed-off-by: James Chapman <jchapman@katalix.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

--- a/net/l2tp/l2tp_eth.c
+++ b/net/l2tp/l2tp_eth.c
@@ -283,7 +283,7 @@ static __net_init int l2tp_eth_init_net(
 	return 0;
 }
 
-static __net_initdata struct pernet_operations l2tp_eth_net_ops = {
+static struct pernet_operations l2tp_eth_net_ops = {
 	.init = l2tp_eth_init_net,
 	.id   = &l2tp_eth_net_id,
 	.size = sizeof(struct l2tp_eth_net),
